docs/examples/MondrianTreeRegressor.md

```diff
@@ -0,0 +1,273 @@
+# Intuition behind MondrianTrees
+
+This example provides an intuition behind the MondrianTreeRegressor. In particular, the key differences between an extremely randomized tree and explanations of the tree construction and prediction will be higlighted.
+
+
+```python
+# Import necessary stuff
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.tree import ExtraTreeRegressor
+from skgarden import MondrianTreeRegressor
+from itertools import cycle
+%matplotlib inline
+```
+
+##  Extremely Randomized Tree - Recap
+
+In a tree-based algorithm, the decision rule at every node is constructed by considering a set of candidate split points and that candidate split point that minimizes a certain impurity (generally the mse in case of continuous values) is chosen. In an extremely randomized tree, the number of candidate splits `S` are limited to `n_features` and each candidate split `S[i]` is drawn unifomly from the bounds `(l_f[i], u_f[i])`.
```

Comment:
also, i would rather say that we aim to "maximize the decrease of impurity in the resulting child nodes (generally the reduction of variance in case of continuous output values)"